react/features/conference/middleware.ts

```diff
@@ -113,5 +144,117 @@ function _conferenceJoined({ dispatch, getState }: IStore) {
         getState
     });
 
+    if (!intervalId) {
+        intervalId = setInterval(() =>
+            _maybeDisplayCalendarNotification({
+                dispatch,
+                getState
+            }), 10 * 1000);
+    }
+
     dispatch(showSalesforceNotification());
 }
+
+/**
+ * Periodically checks if there is an event in the calendar for which we
+ * need to show a notification.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @private
+ * @returns {void}
+ */
+function _maybeDisplayCalendarNotification({ dispatch, getState }: IStore) {
+    const state = getState();
+
+    const calendarEnabled = isCalendarEnabled(state);
+    const { events: eventList } = state['features/calendar-sync'];
+    const { locationURL } = state['features/base/connection'];
+    const { reducedUI } = state['features/base/responsive-ui'];
+
+    const currentConferenceURL
+        = locationURL ? getURLWithoutParamsNormalized(locationURL) : '';
+    const ALERT_MILLISECONDS = 5 * 60 * 1000;
+    const now = Date.now();
+
+    let eventToShow;
+
+    if (eventList?.length) {
```

Comment:
Do we need this check? If the array is empty it should be fine no?

Reply:
I need to check this, but I think it should be fine.